libs/composer/README.md

```diff
@@ -19,26 +30,67 @@
 
 - Run "composer install --no-dev"
 - Add all files to ILIAS git-repository and commit
-- Run "composer install" to reinstall dev-dependencies and build autoload.php
-- ignore all dependencies installed now (which are dev-requirements only)
-- add and commit changes in libs/.gitgnore and autoload.php etc.
 
-## Dependencies for development
+### Dependencies for development
 - Add a new library using composer, e.g. "composer require --dev phpunit/phpunit" 
-- Ignore all directories which are added by installation (uncomment existing)
-- commit changes in autoload.php etc.
+- Do not add the installed dependencies in /libs/composer/vendor to the repository. 
+- Commit changes of composer.json and composer.lock 
 
-## Update a single dependency
+### Update a single dependency
 - Search the name of dependency you like to update.
-- Update by using "composer update --no-dev <DEPENDENCIE_NAME>"
-- Commit all changes in composer.lock and the vendor folder
+- Update by using "composer update --no-dev <DEPENDENCY_NAME>"
+- Commit all changes in composer.lock, composer.json and the vendor folder
+
+### Remove a dependency
+#### Production
+A production dependency can be removed with the following command:
+```bash
+composer remove <DEPENDENCY_NAME>
+```
+Afterwards all changes should be committed.
+ 
+#### Development
+A development dependency can be removed with the following command:
+```bash
+composer remove --dev <DEPENDENCY_NAME>
+```
+Afterwards all changes should be committed.
 
 # Updating Class-Map
 The composer classmap holds information (besides the PHP-dependencies) on all ILIAS-classnames and 
-provied autoloading for them. There is no need to require_once or include_once ILIAS-Classes 
+provides autoloading for them. There is no need to require_once or include_once ILIAS-Classes 
 anymore.
-Whenever you introduce a new ILIAS-Class in /Services\/* or /Modules\/*, the classmap should be 
-updated using 
+Whenever a new ILIAS-Class in /Services\/* or /Modules\/* is introduced, the classmap should be 
+updated with the following command. 
 ```bash
 $ composer dump-autoload
-```
\ No newline at end of file
+```
+
+# Create library patch
```

Comment:
I would propose to disencourage that behaviour somehow, e.g. by putting a "has to be confirmed by JF" or "... TB".